<a href="https://colab.research.google.com/github/Varshan-M/Flyrank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

## 1. My Lane (or Freestyle) and Why

I chose the **Refresh / Content Opportunity Scoring** lane because content teams often have many pages but limited time and resources to update them. Rather than deciding manually, I want to build a ranking system that identifies which pages should be reviewed first based on historical search performance and engagement signals. This project focuses on supporting better content refresh decisions using data-driven evidence instead of intuition.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

## 2. The Question: Decision, Action, Cost of a Wrong Call

**Decision:**  
Which content pages should be prioritized for refresh to maximize the potential improvement in search performance?

**Who acts and what do they do?**  
SEO specialists or content editors use the ranked recommendations to decide which pages to review and update first. They can improve content quality, refresh outdated information, optimize keywords, or enhance user experience.

**Cost of a wrong recommendation:**  
If low-impact pages are recommended, time and resources may be spent on updates that provide little benefit, while high-potential pages continue to miss opportunities for improved visibility and engagement. The output is intended to support decision-making, not replace human judgment.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
!git clone https://github.com/Varshan-M/Flyrank-ML-Internship.git

Cloning into 'Flyrank-ML-Internship'...
remote: Enumerating objects: 110, done.
remote: Counting objects: 100% (110/110), done.
remote: Compressing objects: 100% (81/81), done.
remote: Total 110 (delta 25), reused 81 (delta 13), pack-reused 0 (from 0)
Receiving objects: 100% (110/110), 1.84 MiB | 14.50 MiB/s, done.
Resolving deltas: 100% (25/25), done.


In [6]:
!find Flyrank-ML-Internship -name "*.csv"

Flyrank-ML-Internship/outputs/refresh_queue_sample.csv
Flyrank-ML-Internship/data/raw/content_refresh_anonymized.csv


In [7]:
import pandas as pd

df = pd.read_csv("Flyrank-ML-Internship/data/raw/content_refresh_anonymized.csv")

print("Total content items:", len(df))
print("Number of clients:", df["client_id"].nunique())
print("Average CTR (%):", round(df["ctr"].mean(), 2))
print("Average impressions (90d):", round(df["impressions_90d"].mean(), 2))
print("Average search position (excluding 0):",
      round(df.loc[df["avg_position"] != 0, "avg_position"].mean(), 2))

Total content items: 30000
Number of clients: 32
Average CTR (%): 0.51
Average impressions (90d): 5200.37
Average search position (excluding 0): 17.03


## 3. Quick Look at the Data (2–3 Real Numbers)

The starter dataset contains **30,000** content items from **32** anonymized clients, providing a diverse set of search performance data.

The average click-through rate (CTR) is **0.51%**, the average number of impressions over the last 90 days is **5,200.37**, and the average search position (excluding pages with no position data) is **17.03**.

These numbers indicate that search visibility and engagement vary across content items, making it worthwhile to prioritize which pages should be refreshed first instead of treating every page equally.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful Words: What I Can and Can't Claim

This project provides **decision support** by identifying pages that appear to have higher content refresh opportunities based on historical search performance and engagement data. The results describe **observed patterns** in the available dataset and may indicate useful directional trends for prioritizing content updates.

This project **does not** prove that refreshing a page will cause higher rankings, more traffic, or better engagement. It also does **not** predict or explain Google's ranking algorithm. Any recommendations should be used alongside human judgment and additional business context.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## 5. Self-Check

- ✅ Every required section has been completed with both markdown explanations and supporting code.
- ✅ The notebook runs from top to bottom without errors.
- ✅ No client names, URLs, or private search queries are included.
- ✅ All claims use careful language and are presented as observed, measured, directional, or decision-support findings.
- ✅ The notebook has been saved and committed to my repository under `work/notebooks/`.